src/backend/src/services/teams.services.ts

```diff
@@ -162,4 +162,59 @@ export default class TeamsService {
 
     return teamTransformer(updateTeam);
   }
+
+  /**
+   * Update the given teamId's team's leads
+   * @param submitter a user who's making this request
+   * @param teamId a id of team to be updated
+   * @param userIds a array of user Ids that replaces team's old leads
+   * @returns an updated team
+   * @throws if the team is not found, the submitter has no priviledge, or any user from the given userIds does not exist
+   */
+  static async setTeamLeads(submitter: User, teamId: string, userIds: number[]): Promise<Team> {
+    const team = await prisma.team.findUnique({
+      where: { teamId },
+      ...teamQueryArgs
+    });
+
+    const teams = await prisma.team.findMany();
+
+    const newLeads = await getUsers(userIds);
+
+    if (!team) throw new NotFoundException('Team', teamId);
+
+    if (!isAdmin(submitter.role) && submitter.userId !== team.headId) {
+      throw new AccessDeniedException('You must be an admin or the head to update the lead!');
+    }
+
+    if (newLeads.map((lead) => lead.userId).includes(team.headId)) {
+      throw new AccessDeniedException('The lead cannot be the head of the team!');
+    }
+
+    if (team.members.map((member) => member.userId).some((memberId) => userIds.includes(memberId))) {
+      throw new AccessDeniedException('The lead cannot be a member of the team!');
+    }
+
+    if (teams.map((team) => team.headId).some((teamHeadId) => newLeads.map((lead) => lead.userId).includes(teamHeadId))) {
+      throw new AccessDeniedException('A teams lead must not be a head of another team!');
```

Comment:
we should also verify that none of the new leads are a lead on another team

Reply:
I disagreee